### DISCUSSION TITLE
Whats the use of Embeding this in a web page?

### DISCUSSION THREAD

**Discussion by Darin755** [jeremyckahn/chitchatter#184] [General]
So I have been playing around with this and I am trying to figure out if I can find a neat use for this project. I see that it is embedable which is cool but I can't find an actual reason that it would be useful. If there was persistence it would make sense but why would you just embed something that doesn't store any messages long term? 

There also is an issue of liability. I don't want to embed something that can be used to share CSAM or be used in a way that I would find unethical.

**Comment by jeremyckahn** [comment]
> why would you just embed something that doesn't store any messages long term?

This is a question that I'm mostly leaving up to others to answer. I personally needed this functionality to embed a chat into https://github.com/jeremyckahn/farmhand, a game project of mine. Transcript persistence isn't important for my use case (in fact it would be better to not have it for the liability concerns you raised), so I could see others having similar use cases.

I can imagine a number of use cases where a website operator would want to easily enable chat on their page without worrying about message log management or cost. I'm thinking along the lines of hobby projects with no budget or perhaps support websites for vulnerable communities.

It's worth mentioning that I'm developing an SDK for Chitchatter embeds in #183 that will enable greater customization of the chat embed. That should be available soon.

> There also is an issue of liability. I don't want to embed something that can be used to share CSAM or be used in a way that I would find unethical.

In that case, Chitchatter may not be the right fit for your needs. This project is designed to give as much freedom to the end user as possible. Users can choose to use that freedom for either good things or bad things. Any illegal activity is a matter for law enforcement to handle as I have no visibility into it.

As for liability: Chitchatter does not store conversation data, and conversation data is never exposed to a server. So, it's not clear to me what sort of liability you might have as a website operator that incorporates a Chitchatter embed.

**Reply by Darin755** [reply]
Interesting, thanks for the reply
